### DISCUSSION TITLE
Wahoo kickr bike and ifit incline?

### DISCUSSION THREAD

**Discussion by mtluczek** [cagnulein/qdomyos-zwift#2207] [General]
Hi,

I'd like to ask if Kickbike v3 will work with iFit to adaptively change the incline and resistance. If not, how much would it be necessary to contribute? Do you think that would be something difficult to work on? What'd be required. I'm a programmer as well.

**Comment by cagnulein** [comment]
I guess it will already work at 99%

Give it a try and report back!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 11 mar 2024 alle 00:17 mtluczek ***@***.***> ha
scritto:

**Reply by mtluczek** [reply]
I've just found this info: https://support.wahoofitness.com/hc/en-us/articles/204281764-Trainer-Control-Modes-for-KICKR-CORE-SNAP-or-BIKE-in-the-Wahoo-app 

They say: Due to brake design, Resistance mode is not available for BIKE and BIKE shift models.

Wondering how are those iFit commands with resistance interpreted by the bike. I hope I won't break anything :) maybe somehow it could be possible to translate the iFit cadence/resistance into some SIM mode resistance?

**Reply by mtluczek** [reply]
or is it already translating it to SIM mode resistance?

**Reply by cagnulein** [reply]
@mtluczek i have already that possibility in the code. did you try it?

**Reply by mtluczek** [reply]
Hi, I bought your iOS app to try it and it seems I need Android phone QZ app and not iOS one or vice versa. I'm trying to install virtualbox Android machine with BT support.

**Comment by cagnulein** [comment]
You need to enable the virtual ifit and to rename your android device to
"I_EL" and reboot your phone

let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 13 mar 2024 alle 23:51 mtluczek ***@***.***> ha
scritto:

**Reply by mtluczek** [reply]
What's if it's the other way around? Shall I change iOS device name as well?

**Comment by cagnulein** [comment]
nope on ios qz changes it automatically when you enable the virtual ifit
setting

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 14 mar 2024 alle ore 10:05 mtluczek ***@***.***>
ha scritto:

**Reply by mtluczek** [reply]
OK, I connected and resistance seem to work, I can see the speed also, but incline is not working and not even showing in ifit. I figured out that incline control doesn't work at all in QZ app on iOS without any iFit - I try to change the incline but nothing changes. Should I turn on the incline control somewhere?

**Reply by cagnulein** [reply]
got the issue, it's because i implemented on the resistance only now (99% of the bikes have only resistance). I will be glad to add it! Please create a ticket with a current debug log from your scenario and I will do it! https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

**Reply by mtluczek** [reply]
I mean it works on the bike - then the bike goes into lock mode when I start to change incline on the handlebar. But in QZ there is this icon with incline change but it doesn't move up/down at all - remains at 0 without any control.

**Reply by mtluczek** [reply]
Done - ticket  added. How will the app be updated after you're done?

**Reply by cagnulein** [reply]
on ios via testflight, on android using the beta system :)
